src/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb

```diff
@@ -122,6 +142,9 @@ def delete(instance_id)
         if network_interface[:tags].has_key?('availability_set')
           delete_availability_set(network_interface[:tags]['availability_set'])
         end
+        if network_interface[:tags].has_key?('ephemeral_disk_id')
```

Comment:
Yes. But if the VM is deleted but the network interface is not deleted. We should make sure that the data disk can be removed when retrying.